oxidauth-usecases/src/auth/strategies/oauth2/microsoft/exchange_token.rs

```diff
@@ -0,0 +1,36 @@
+use reqwest::header::CONTENT_TYPE;
+
+use crate::auth::strategies::oauth2::AuthorityParams;
+use oxidauth_kernel::error::BoxedError;
+
+use super::{MicrosoftExchangeTokenReq, MicrosoftExchangeTokenRes};
+
+#[tracing::instrument(name = "microsoft oauth exchange token")]
+pub async fn exchange_microsoft_token(
+    code: String,
+    params: &AuthorityParams,
+) -> Result<String, BoxedError> {
+    let json = MicrosoftExchangeTokenReq {
```

Comment:
Same here -- this struct isn't returned and is short-lived. We can leverage borrows here.

Reply:
I don't actually know how to do this. Because the MicrosoftExchangeTokenReq has derive `Deserialize`, it can't contain borrows as far as I understand?